### ISSUE TITLE
Aliases on interface types do not work

### ISSUE THREAD

**Issue by jroith** [neo4j/graphql#2844]
Aliases are broken on fields that have an interface type.

For example given the schema:

```GraphQL
interface I {
  name: String!
}
type A implements I {
  name: String!
}
type B implements I {
  name: String!
}

type Foo {
  bar: [I!]! @relationship(type: "TEST", direction: IN)
}

```

And the Query

```GraphQL
{
  foos {
    alias1: bar(where: { name: "test"}) {
      name
    }
    alias2: bar(where: { name: "test2"}) {
      name
    }
  }
}

```

This will throw an error that a cypher variable is defined twice. Note that this works for fields that do not have an interface type.
This can presumably be fixed by including the alias name in the cypher query.

**Comment by neo4j-team-graphql** [comment]
Many thanks for raising this bug report @jroith. :bug: We will now attempt to reproduce the bug based on the steps you have provided.

Please ensure that you've provided the necessary information for a minimal reproduction, including but not limited to:
* Type definitions
* Resolvers
* Query and/or Mutation (or multiple) needed to reproduce

If you have a support agreement with Neo4j, please link this GitHub issue to a new or existing Zendesk ticket.

Thanks again! :pray:

**Comment by neo4j-team-graphql** [comment]
Many thanks for raising this bug report @jroith. :bug: We will now attempt to reproduce the bug based on the steps you have provided.

Please ensure that you've provided the necessary information for a minimal reproduction, including but not limited to:
* Type definitions
* Resolvers
* Query and/or Mutation (or multiple) needed to reproduce

If you have a support agreement with Neo4j, please link this GitHub issue to a new or existing Zendesk ticket.

Thanks again! :pray:

**Comment by Liam-Doodson** [comment]
Hi @jroith! The type definitions and query that you provided seem to be incomplete and do not work together. As a result, I've been unable to confirm the bug. Can you please provide updated type definitions, query and cypher that I can use to reproduce this

**Comment by jroith** [comment]
I have updated it with what I think was more or less missing, but haven't tested this example in isolation. I'll give it a try later.

**Comment by jroith** [comment]
I have now tested this query and it the result is:

```
Variable `this_bar` already declared in outer scope (line 29, column 57 (offset: 598))
"    RETURN { __resolveType: "A", name: this_A.name } AS this_bar"
```

Here is a fully example to reproduce this (adapt credentials): **index.js**

```Javascript
const { gql, ApolloServer } = require("apollo-server");
const { Neo4jGraphQL } = require("@neo4j/graphql");
const neo4j = require("neo4j-driver");

const typeDefs = gql`
interface I {
  name: String!
}
type A implements I {
  name: String!
}
type B implements I {
  name: String!
}

type Foo {
  bar: [I!]! @relationship(type: "TEST", direction: IN)
}
`;

const driver = neo4j.driver(
  "bolt://localhost:7687/neo4j",
  neo4j.auth.basic("neo4j", "test")
);

const neoSchema = new Neo4jGraphQL({ typeDefs, driver });

neoSchema.getSchema().then((schema) => {
    const server = new ApolloServer({
        schema: schema
    });

    server.listen().then(({ url }) => {
        console.log(`GraphQL server ready on ${url}`);
    });
});
```

Install and run server:

```Sh
yarn init
yarn add apollo-server @neo4j/graphql neo4j-driver graphql
node index.js
```

Run query:

```Sh
curl -X "POST" "http://localhost:4000/graphql" \
     -H 'Content-Type: application/json; charset=utf-8' \
     -d $'{
	"query": "{foos{alias1:bar(where:{name:\\"test\\"}){name}alias2:bar(where:{name:\\"test2\\"}){name}}}",
	"variables": {}
}'
```

**Comment by neo4j-team-graphql** [comment]
We've been able to confirm this bug using the steps to reproduce that you provided - many thanks @jroith! :pray: We will now prioritise the bug and address it appropriately.

**Comment by neo4j-team-graphql** [comment]
This bug report has been assigned high priority to fix. If you wish to contribute a fix, please branch from `master` and submit your PR with the base set to `master`. Thanks!

**Comment by MacondoExpress** [comment]
Thank you for the report [jroith](https://github.com/jroith), this is already closed by #2879 and will be released soon!
